GameEngine/Services/Managers/PairManager/PairManager.cs

```diff
@@ -14,20 +16,19 @@ public class PairManager : IPairManager
         public Pair? Pair { get; private set; }
 
         /// <summary>
-        /// Creates new pair for current animal, searching opposite free animal near by this animal. 
+        /// Creates new pair for current pairable object, searching opposite free pairable object near by this object. 
         /// </summary>
-        /// <param name="animals">Animals.</param>
-        public void CreatePair(List<Animal> animals)
+        /// <param name="pairables">Pairables.</param>
+        public void CreatePair(List<IItem> pairables)
         {
-            foreach (var currentAnimal in animals)
+            foreach (var currentAnimal in pairables.Cast<Animal>())
             {
                 if (!currentAnimal.IsPaired)
                 {
-                    List<Animal> animalsAround = Helper.LookAround(currentAnimal, animals).ToList();
+                    List<Animal> animalsAround = Helper.LookAround(currentAnimal, pairables).Cast<Animal>().ToList();
 
                     List<Animal> freeOppositeSexAnimalsAround = animalsAround
                         .Where(animal => !animal.Sex.Equals(currentAnimal.Sex) && !animal.IsPaired && animal.GetType() == currentAnimal.GetType())
```

Comment:
Sex of animal. Was that required?

Reply:
It was not required, but seemed logical.